### PR TITLE
glib-macros: Port clone failure tests to trybuild

### PR DESCRIPTION
**Pull request by jf2048**
cc @GuillaumeGomez 

Saves about 65 lines of code

### CODE REVIEW

**Comment by jf2048** [comment]
A downside with this approach: Eventually I would like to make the error messages more specific, e.g. the "Can't use self" error should point only to the `self` token. Making those kind of changes will break this approach because of the lack of fuzzy matching.

File: glib-macros/tests/clone.rs [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+     "Expected `move` after `async`, found `{`"),
+    ("clone!(@strong v => move || async println!(\"foo\");)",
+     "Expected `move` after `async`, found `println`"),
+    ("clone!(@strong v => move || async move println!(\"foo\");)",
+     "Expected block after `| async move`"),
+];
+
+#[test]
+fn clone_failures() {
+    let exe = std::env::current_exe().unwrap();
```
Comment (GuillaumeGomez) [inline glib-macros/tests/clone.rs:76]: It'll create a lot of files which will remain after the tests are done. :-/

**Comment by GuillaumeGomez** [comment]
If you don't mind, I'll cherry-pick your commit and give a try to my `trybuild` extension to have inline tests instead of handling the file part ourselves.

**Comment by GuillaumeGomez** [comment]
By using your commit, I was able to reduce the diff quite a lot. I'll send a PR with your commit in it shortly.

**Pull request closed without merging**